src/Controls/src/Core/Platform/ModalNavigationManager/ModalNavigationManager.Android.cs

```diff
@@ -58,17 +75,40 @@ internal bool NavAnimationInProgress
 			}
 		}
 
-		public Task<Page> PopModalAsync(bool animated)
+		Task RemoveModalPage(Page page)
 		{
-			Page modal = _navModel.PopModal();
-			var source = new TaskCompletionSource<Page>();
+			_platformModalPages.Remove(page);
 
-			var modalHandler = modal.Handler as IPlatformViewHandler;
+			var modalHandler = page.Handler as IPlatformViewHandler;
 			if (modalHandler != null)
 			{
 				ModalContainer? modalContainer = null;
+				for (int i = 0; i <= GetModalParentView().ChildCount; i++)
+				{
+					if (GetModalParentView().GetChildAt(i) is ModalContainer mc &&
+						mc.Modal == page)
+					{
+						modalContainer = mc;
+					}
+				}
+
+				modalContainer?.Destroy();
+			}
+
+			return Task.CompletedTask;
+		}
 
+		Task<Page> PopModalPlatformAsync(bool animated)
+		{
+			Page modal = CurrentPlatformModalPage;
+			_platformModalPages.Remove(modal);
+
+			var source = new TaskCompletionSource<Page>();
 
+			var modalHandler = modal.Handler as IPlatformViewHandler;
+			if (modalHandler != null)
```

Comment:
Why don't you use the is operator and merge the two statements?
```suggestion
			var modalHandler = modal.Handler as IPlatformViewHandler;
			if (modal.Handler is IPlatformViewHandler modalHandler)
```